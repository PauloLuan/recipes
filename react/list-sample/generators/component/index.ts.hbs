export * from './{{pascalCase componentName}}.component'
