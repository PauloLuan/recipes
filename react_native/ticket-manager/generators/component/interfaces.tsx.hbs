import { I{{capitalize componentName}}Props } from 'native-base'

export interface {{capitalize componentName}}Props extends I{{capitalize componentName}}Props {
  testId?: string
}
