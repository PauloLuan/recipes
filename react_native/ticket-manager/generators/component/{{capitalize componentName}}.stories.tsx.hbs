import { storiesOf } from '@storybook/react-native'
import React from 'react'
import { {{capitalize componentName}} } from './{{capitalize componentName}}'

storiesOf('Components/{{capitalize componentName}}', module).add('default: ', () => <{{capitalize componentName}} />)
