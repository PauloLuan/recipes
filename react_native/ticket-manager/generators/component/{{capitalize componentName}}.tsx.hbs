import React from 'react'
import { I{{capitalize componentName}}Props, {{capitalize componentName}} as NativeBase{{capitalize componentName}} } from 'native-base'

export interface {{capitalize componentName}}Props extends I{{capitalize componentName}}Props {
  testId?: string
}

export const {{capitalize componentName}} = ({
  testId = '{{kebabCase componentName}}-id',
  ...props
}: {{capitalize componentName}}Props) => {
  return (
    <NativeBase{{capitalize componentName}} testID={testId} {...props}></NativeBase{{capitalize componentName}}>
  )
}
