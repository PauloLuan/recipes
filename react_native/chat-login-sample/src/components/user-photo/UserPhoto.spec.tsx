import { render, waitFor } from '@testing-library/react-native'
import React from 'react'
import { UserPhoto } from './UserPhoto.component'

// import theme from '../../styles/Theme'

describe('<UserPhoto />', () => {
  const testId = 'component-user-photo-id'

  it('should render the UserPhoto component', async () => {
    const { getByTestId, queryByTestId, toJSON } = render(
      <UserPhoto testId={testId} />
    )
    await waitFor(() => expect(queryByTestId(testId)).toBeTruthy())
    expect(toJSON()).toMatchSnapshot()
  })
})
