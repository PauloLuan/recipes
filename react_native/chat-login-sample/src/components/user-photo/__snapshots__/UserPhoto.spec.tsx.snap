// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserPhoto /> should render the UserPhoto component 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "blue",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": 60,
        "width": 85,
      },
    ]
  }
  testID="component-user-photo-id"
>
  Body...
</View>
`;
